Add IQaudIO Sound Card support for Raspberry Pi

iqaudio opened the pull request:
Request to merge into Raspberry Pi Linux, Would like to merge into Florian's ASOC fork too.

Commit:
Add IQaudIO Sound Card support for Raspberry Pi

popcornmix commented:
@koalo any comments?

koalo commented:
I think it is ok. It could have been split up into multiple commits and the "NOT USED" lines could be removed (together with the whole snd_rpi_iqaudio_dac_init function), but all in all it looks good.

hmbedded committed:
Fix volsw_range functions so SOC_DOUBLE_R_RANGE_TLV works.

This is so that the correct rabge of values as specified
with the SOC_DOUBLE_R_RANGE_TLV macro are sent to the
hardware for both the normal and invert cases.

hmbedded committed:
Use a range macro for Playback Volume.

This allows limiting the output gain to avoid clipping in the
DAC ouput stages.

amtssp commented:
I hope you will add this to the 3.13.y kernel as well.
What is the time schedule for this?
Thanks

Commit:
fix soc-core's inverse range and let IQaudIO DAC use this fixed SOC_DOUBLE_R_RANGE_TLV support
